sw/device/lib/dif/dif_aes.h

```diff
@@ -255,14 +305,17 @@ dif_result_t dif_aes_reset(const dif_aes_t *aes);
  *
  * @param aes AES state data.
  * @param transaction Configuration data.
- * @param iv AES Initialisation Vector. The iv may not be used for some modes,
- * see `dif_aes_mode_t` for more details.
+ * @param key Encryption/decryption key when `kDifAesKeySoftwareProvided`, must
+ * be `NULL` otherwise.
+ * @param iv Initialization vector when the mode isn't `kDifAesModeEcb`, must be
+ * `NULL` otherwise.
```

Comment:
These comments are wrong: The implementation just checks whether key/IV are not NULL in case they are used but the comments say that they aren't used, they must be NULL. Basically, you can just change "must" to "can".